### PR TITLE
T1 added

### PR DESCRIPTION
**Pull request by vkrekic-endava**

### CODE REVIEW

File: src/main/java/com/endava/pages/HomePage.java [hunk 1]
```diff
@@ -35,15 +36,26 @@ public MenuPage openMenu() {
 	}
 
 	/**
-	 * finds the down arrow element and clicks on it	 * 
+	 * Opens AgilePage and instantiate AgilePage object
+	 * @author Vladimir Krekic
+	 * @return AgilePage
+	 */
+	public AgilePage openAgilePage() {
+		driver.findElement(this.agileItem).click();
```
Comment (endavamarko) [inline src/main/java/com/endava/pages/HomePage.java]: We should assert is element present and print message if element is not present, and than to click. If we don't check this and method fails, we don't know is it not present or not clickable. Example with pseudo-code:
log.debug("opis metode");
isElementPresent(locator,Time.LONG);
locator.click();
There are many ways to do that:
Most simple is check is webelements array empty
Other methods:
public static void waitForElementToAppear(Driver driver, By selector, long timeOutInSeconds, String timeOutMessage) {
try {
WebDriverWait wait = new WebDriverWait(driver, timeOutInSeconds);
wait.until(ExpectedConditions.visibilityOfElementLocated(selector));
} catch (TimeoutException e) {
throw new IllegalStateException(timeOutMessage);
}
}
Method with displayed()
public boolean isElementPresent(WebElement locator) {
try {
return locator.isDisplayed();
}
catch (final Exception e) {
return false;
}
}

File: src/test/java/com/endava/TestAgilePage.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @Test
+    public void testAgileItemActiveInDAAMenu() {
+        homePage.open();
+        menuPage = homePage.openMenu();
+        new WebDriverWait(menuPage.driver, 5)
+                .until(ExpectedConditions.visibilityOfElementLocated(menuPage.getNavigationList()));
+        agilePage = homePage.openAgilePage();
+        new WebDriverWait(agilePage.driver, 5)
+                .until((ExpectedConditions.attributeContains(agilePage.getAgileSecondaryNav(), "class", "active")));
+        log.info("testAgileItemActiveInDAAMenu()");
```
Comment (BoskoNikolic) [inline src/test/java/com/endava/TestAgilePage.java]: log treba da objasni akciju ne samo da ispise ime metode.

File: src/test/java/com/endava/TestAgilePage.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * @author Vladimir Krekic
+     * Test validates AGILE menu item is active in DIGITAL - AGILE - AUTOMATION menu
+     */
+    @Test
+    public void testAgileItemActiveInDAAMenu() {
+        homePage.open();
+        menuPage = homePage.openMenu();
+        new WebDriverWait(menuPage.driver, 5)
```
Comment (endavamarko) [inline src/test/java/com/endava/TestAgilePage.java]: WebDriverWait ne treba u test metodama, nego napraviti metodu u Utils-u I samo proslediti parametre

File: src/test/java/com/endava/TestAgilePage.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    public void setUp(String browser) {
+        homePage = Utils.setUpWebBrowser(browser);
+        log.info("setUp()");
+    }
+
+    /**
+     * @author Vladimir Krekic
+     * Test validates AGILE menu item is active in DIGITAL - AGILE - AUTOMATION menu
+     */
+    @Test
```
Comment (BoskoNikolic) [inline src/test/java/com/endava/TestAgilePage.java]: Gde ti je Assert u testu?

File: src/test/java/com/endava/TestHomePage.java [hunk 5]
```diff
@@ -39,10 +41,12 @@ public void setUp(String browser) {
 	}
 
 	/**
+	 * @author Jelena Corak
+	 * @author Goran.Kukolj
```
Comment (BoskoNikolic) [inline src/test/java/com/endava/TestHomePage.java]: Samo originalni autor

File: src/test/java/com/endava/TestHomePage.java [hunk 6]
```diff
@@ -55,7 +59,9 @@ public void testHomePageIsOpened() {
 	}
 
 	/**
-   * @author Vladimir Krekic
+	 * @author Jelena Corak
+	 * @author Goran.Kukolj
```
Comment (BoskoNikolic) [inline src/test/java/com/endava/TestHomePage.java]: Samo originalni autor

File: src/test/resources/testng.xml [hunk 7]
```diff
@@ -4,6 +4,7 @@
         <parameter name="browser" value="chrome"/>
         <classes>
             <class name="com.endava.TestHomePage" />
+            <class name="com.endava.TestAgilePage"/>
```
Comment (BoskoNikolic) [inline src/test/resources/testng.xml]: razmak pre />

**Comment by vkrekic-endava** [comment]
closed  and new one made

**Pull request closed without merging**